torch/_inductor/lowering.py

```diff
@@ -5108,6 +5108,64 @@ def log_add_exp_helper(a, b):
     return result
 
 
+@register_lowering(aten.cummax, type_promotion_kind=None)
+def cummax(x, axis=None):
+    if len(x.get_size()) == 0:
+        assert axis in [0, -1]
+        return clone(x), torch.empty_like(x, dtype=torch.int64)
+
+    def combine_fn(value1, index1, value2, index2):
+        gt = ops.gt(value1, value2)
+        return (ops.where(gt, value1, value2), ops.where(gt, index1, index2))
```

Comment:
This is missing nan handling and the tie breaker which handles the case where `value1 == value2`. Probably better to just use `get_reduction_combine_fn("argmax", dtype)`